### PR TITLE
API to set viewport for a scene

### PR DESCRIPTION
**Pull request by deepakrawat22**
API to set viewport. Usage:

mainScene.getMainCameraRig().setViewport(x, y, width, height);

GearVRf-DCO-1.0-Signed-off-by: Deepak Rawat
deepak.rawat@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Ok, how about instead of punching this hole in the APIs we add viewportX, viewportY, viewportWidth and viewportHeight as configuration parameters to eyeBufferParms?

**Comment by deepakrawat22** [comment]
Having the API in getMainCameraRig() allows to change the viewport at run-time, although can't say if its a useful feature to change viewport during run-time.

**Comment by liaxim** [comment]
Well, where does this API really belong? It is something highly specific so until we know why we want it dynamic I'd say make it a static, somewhat hidden, configuration. Would your vuforia changes work across devices with a static config?

**Comment by deepakrawat22** [comment]
Yes vuforia changes works well with static config. Ok will move the viewport to config file. Thanks.

**Comment by deepakrawat22** [comment]
Set viewport in config file

File: GVRf/Framework/framework/src/main/jni/oculus/activity.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ class GVRActivity
 
     bool updateSensoredScene();
     void setCameraRig(jlong cameraRig);
+    void setViewport(int x_, int y_, int width_, int height_);
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/jni/oculus/activity.h]: Unused. Remove?

**Comment by liaxim** [comment]
Will merge in the afternoon

**Pull request merged**